### PR TITLE
additions and edits to documentation

### LINKED ISSUE #1542: Update Navigation System doc to reflect current recommendations
Update https://www.patternfly.org/v4/design-guidelines/usage-and-behavior/navigation-system to reflect most current navigation mockups and guidance. 

* Since we are not supporting the Context Selector placed in the masthead as an "official" PatternFly design pattern, want to remove references to it from the page.

* Check examples against current dark nav variants.

### LINKED ISSUE #1608: Add design guidelines for inline edit
Introduce a new design guideline document for inline editing applied at the field, row, or page level. 

Design documentation: https://docs.google.com/document/d/15aOWgpkd0Zp0hvbKM4jfG6Jf00FHjY_f92bkosbo1Ds/edit?usp=sharing

### LINKED ISSUE #1714: Banner design guidelines
Introduce guidelines for usage of the Banner component per designs here: https://github.com/patternfly/patternfly-design/issues/765

### LINKED ISSUE #1842: Add Skeleton Loading to Data Loading guidelines
Add guidance for Skeleton Loading to the documentation here: https://www.patternfly.org/v4/design-guidelines/usage-and-behavior/data-loading  Be sure to discuss when to use skeleton loading vs spinners or progress bars. More background in the design issue here: https://github.com/patternfly/patternfly-design/issues/766

### PR DESCRIPTION
**Pull request by mmenestr**
Made edits to the following documentation:

**Components**
Accordion
Badge
Banner (Fixes #1714  )
Breadcrumb
Checkbox
Chip
Chip group
Context selector
Data list
Drawer
Form select
Hint
Inline edit (Fixes #1608  )
Input group
Navigation (Fixes #1542 )
Popover
Radio
Select
Skeleton (Fixes #1842  )
Spinner
Switch
Tab content
Table
Text
Text area
Text input 
Tile
Tooltip

**Content**
About

**Contribute**
Design

### CODE REVIEW

File: packages/v4/src/content/design-guidelines/components/badge/badge.md [hunk 1]
```diff
@@ -2,4 +2,11 @@
 id: Badge
 section: components
 ---
-A **badge** is used to annotate other information like a label or an object name. Badges are typically used to reflect a count, e.g. number of object, number of events, number of unread, etc.
+A **badge** is used to annotate other information like a label or an object name. 
+
+## Usage
+
+Badges are typically used to reflect a count, like number of object, number of events and number of unread, for example.
```
Comment (mcarrano) [inline packages/v4/src/content/design-guidelines/components/badge/badge.md:9]: typos: change "object" to "objects"; "unread" to "unread items"

File: packages/v4/src/content/design-guidelines/components/breadcrumb/breadcrumb.md [hunk 2]
```diff
@@ -2,4 +2,21 @@
...
+## Usage
+
+<img src="./img/breadcrumb.png" alt="Example of breadcrumbs" width="990"/>
+
+Use breadcrumbs in addition to your global navigation to display a user's location in the application.
+
+### Guidelines
+* Use breadcrumbs when the primary global navigation items are not always exposed.
+* Never use breadcrumbs as a replacement for global navigation.
+* Never use breadcrumbs to map a user’s personal journey through a website as it may not reflect where the page actually lives.
```
Comment (mcarrano) [inline packages/v4/src/content/design-guidelines/components/breadcrumb/breadcrumb.md:16]: I understand what is being said here, but I might reword. Perhaps say, "Breadcrumbs in PatternFly are intended to show the location of a page in the site hierarchy, and are not intended to map the user's path through the application. "

File: packages/v4/src/content/design-guidelines/components/drawer/drawer.md [hunk 3]
```diff
@@ -3,3 +3,13 @@ id: Drawer
 section: components
 ---
 The **Drawer** is a sliding panel that enters from the right edge of the viewport. It can be configured to either overlay content on a page or create a sidebar by pushing that content to the left.
+
+## Usage
+
+Drawers can be used to store and show more information than immediately presented on a page. A drawer with more information is usually prompted by a user click. Upon the click, the drawer comes out from the right side of the page (right to left) and shows more information, which is usually relevant to a whole page.
```
Comment (mcarrano) [inline packages/v4/src/content/design-guidelines/components/drawer/drawer.md:9]: We are currently adding a horizontal drawer in the next release so may want to update this.

File: packages/v4/src/content/design-guidelines/components/inline-edit/inline-edit.md [hunk 4]
```diff
@@ -3,5 +3,96 @@ id: Inline edit
 section: components
 ---
 
-We’re working on some guidelines for this component.
-If you have any questions in the meantime, feel free to [get in touch with us.](/get-in-touch)
+Users can use inline edit to switch between a read-only view and an edit view of [description lists](/components/description-list), page text elements, or [tables](/components/table) as needed. In other words, inline editing saves the user from opening a modal window or redirecting to a separate edit page in which the user easily loses the context from the current view. 
+Usage
```
Comment (mcarrano) [inline packages/v4/src/content/design-guidelines/components/inline-edit/inline-edit.md:7]: Does this need to be a heading?

File: packages/v4/src/content/design-guidelines/components/skeleton/skeleton.md [hunk 5]
```diff
@@ -3,5 +3,17 @@ id: Skeleton
 section: components
 ---
 
-We’re working on some guidelines for this component.
-If you have any questions in the meantime, feel free to [get in touch with us.](/get-in-touch)
+A **skeleton** is a type of loading state that shows an example of how data might be populated on a page before it gets populated. For content that may take a long time to load, use a [progress bar](/components/progress/react) in place of a skeleton.
+
+## Usage 
+Skeleton loading should be progressive - it should load static text first, followed by variable elements of the UI.
```
Comment (mcarrano) [inline packages/v4/src/content/design-guidelines/components/skeleton/skeleton.md:9]: I had to read this a few times to understand what it meant. Maybe there's a better way to say this? Perhap's we should say that skeleton loading allows you to expose content incrementally?

**Review by mcarrano — CHANGES_REQUESTED** [review]
@mmenestr First, thanks for doing this. I know it's a tremendous amount of work! I gave this content a quick scan and identified a few minor problems. If needed, I can give this a deeper review next week.

File: packages/v4/src/content/design-guidelines/components/badge/badge.md (hunk 1 above)
Comment (mmenestr) [reply packages/v4/src/content/design-guidelines/components/badge/badge.md:9]: Updated!

File: packages/v4/src/content/design-guidelines/components/breadcrumb/breadcrumb.md (hunk 2 above)
Comment (mmenestr) [reply packages/v4/src/content/design-guidelines/components/breadcrumb/breadcrumb.md:16]: Updated!

File: packages/v4/src/content/design-guidelines/components/drawer/drawer.md (hunk 3 above)
Comment (mmenestr) [reply packages/v4/src/content/design-guidelines/components/drawer/drawer.md:9]: Ok, for now i've removed the "from the right side of the page (right to left)" so it's more flexible

File: packages/v4/src/content/design-guidelines/components/inline-edit/inline-edit.md (hunk 4 above)
Comment (mmenestr) [reply packages/v4/src/content/design-guidelines/components/inline-edit/inline-edit.md:7]: Oh yes good catch!!

File: packages/v4/src/content/design-guidelines/components/skeleton/skeleton.md (hunk 5 above)
Comment (mmenestr) [reply packages/v4/src/content/design-guidelines/components/skeleton/skeleton.md:9]: Updated!

**Comment by mmenestr** [comment]
@mcarrano thorough reviews are always best, but I'm also having Gina review it so two set of eyes should do the trick! Thank you!!

**Review by abigaeljamie — DISMISSED** [review]

**Review by gdoyle1 — CHANGES_REQUESTED** [review]
Found some things that need fixes before merging:
1. **Banner** - inline alert link is linking to input group guidelines
2. **Banner** - One spelling of "Patternfly" that should be "PatternFly"
3. **Data list** - "bulk selection" needs to be linked here /components/table/react-demos ?? there's a bunch of random stuff in that tab....
4. **Form select** - probably don't need to link the "form select" text since it's on the same page just diff tab
5. **Inline edit** - "inline edit animation" image isn't working
6. **Input group** - again not sure if we need to link to "input group documentation" since it's in the other tab

#3 is kinda concerning, not sure how we should approach the content listed in that tab

List of things that may be good to add **in the future**:
1. **Badges** shouldn't be used if selectable tags are needed (instead use labels) - I think this fits in a future issue we already have filed
2. **Context selector** can be placed within the header of a page as well (on white part of the screen) where it changes the context of the data shown on that page

**Comment by mmenestr** [comment]
@gdoyle1 thanks for the thorough review!! That bulk select content under the table react tab is going to be moved to table design guidelines eventually! And I'll add those future things to issei's, thank you!!

**Review by redallen — APPROVED** [review]

**Review by gdoyle1 — APPROVED** [review]
LGTM

**Review by mcarrano — APPROVED** [review]
Looks great. Thanks @mmenestr !

**Pull request merged**